src/tasks/task-joint-posVelAcc-bounds.cpp

```diff
@@ -0,0 +1,491 @@
+      {
+        if(qa[i] < (m_qMin[i] - eps))
+        {
+            if(verbose)
+            {
+              std::cout << "State of joint "<< i <<" is not viable because q[i]< qMin[i] : "
+              <<qa[i] <<"<"<< m_qMin[i] <<std::endl;
+            }
+            m_viabViol[i] = m_qMin[i] - qa[i];
+
+        }
+        if(qa[i] > (m_qMax[i] + eps))
+        {
+            if(verbose){
+              std::cout << "State of joint "<< i <<" is not viable because qa[i]>m_qMax[i] : "
+              << qa[i] <<">"<< m_qMax[i] <<std::endl;
+            }
+            m_viabViol[i] =qa[i]-m_qMax[i];
+
+        }
+        if(std::abs(dqa[i]) > (m_dqMax[i] + eps))
+        {
+            if(verbose)
+            {
+              std::cout << "State (q,dq) :("<<qa[i]<<","<<dqa[i]<<") of joint "
+              << i <<" is not viable because |dq|>dqMax : "<< std::abs(dqa[i]) <<
+              ">"<< m_dqMax[i]<<std::endl;
+            }
+            m_viabViol[i] =std::abs(dqa[i])-m_dqMax[i];
+
+        }
+        double dqMaxViab =   std::sqrt(std::max(0.0, 2*m_ddqMax[i]*(m_qMax[i]-qa[i])));
+        if(dqa[i]>(dqMaxViab+eps))
+        {
+            if(verbose)
+            {
+              std::cout << "State (q,dq,dqMaxViab) :("<<qa[i]<<","<<dqa[i]<<","<<
+              dqMaxViab<<") of joint "<< i <<" is not viable because dq>dqMaxViab : "
+              << dqa[i] <<">"<< dqMaxViab<<std::endl;
+            }
+            m_viabViol[i] =dqa[i]-dqMaxViab;
+
+        }
+        double dqMinViab =  -std::sqrt(std::max(0.0,2*m_ddqMax[i]*(qa[i]-m_qMin[i])));
+        if(dqa[i]<(dqMinViab+eps))
+        {
+            if(verbose)
+            {
+              std::cout << "State (q,dq,dqMinViab) :("<<qa[i]<<","<<dqa[i]<<","<<
+              dqMinViab<<") of joint "<< i <<" is not viable because dq<dqMinViab : "
+              << dqa[i] <<"<"<< dqMinViab<<std::endl;
+            }
+            m_viabViol[i]=dqMinViab-dqa[i];
+        }
+      }
+
+      // if((m_viabViol == Vector::Zero(m_na))&&verbose==true)
+      // {
+      //   // std::cout << "State (q,dq) :("<<qa.transpose()<<","<<dqa.transpose()<<"is viable"<<std::endl;
+      //   std::cout << "State is viable"<<std::endl;
+      // }
+    }
+
+    void TaskJointPosVelAccBounds::computeVelLimits(const Vector& q, bool verbose)
+     {
+        Vector qa = q.tail(m_na);
+        for(int i = 0; i<m_na; i++)
+        {
+        if(verbose)
+        {
+          if(qa[i] < (m_qMin[i] - eps))
+          {
+            std::cerr << "State of joint "<< i <<" is not viable because q[i]<qMin[i] : "
+            << qa[i] <<"<"<< m_qMin[i] <<std::endl;
+          }
+          if(qa[i] < (m_qMax[i] + eps))
+          {
+            std::cerr << "State of joint "<< i <<" is not viable because qa[i]>m_qMax[i] : "
+            << qa[i] <<">"<< m_qMax[i] <<std::endl;
+          }
+          m_dqMaxViab[i] = std::min( m_dqMax[i],  std::sqrt(std::max(0.0,2*m_ddqMax[i]*(m_qMax[i]-qa[i]))));
```

Comment:
This line is inside the `if(verbose)` branch. Is that done on purpose? 
Moreover indentation is off.